python/cudf/cudf/core/column/column.py

```diff
@@ -2230,6 +2231,12 @@ def as_column(
         data = ColumnBase.from_scalar(arbitrary, length if length else 1)
     elif isinstance(arbitrary, pd.core.arrays.masked.BaseMaskedArray):
         data = as_column(pa.Array.from_pandas(arbitrary), dtype=dtype)
+    elif isinstance(arbitrary, pd.DatetimeIndex) and is_datetime64tz_dtype(
```

Comment:
Could you use `isinstance(..., pd.DatetimeTZDtype)`? (`is_datetime64tz_dtype` will be deprecated in pandas 2.1)

Reply:
Done 👍  But we might want to change the rest of the code base in follow-up PRs similarly.